Mercenary UIV

assignedgod opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
This is a new UIV to spice up the common rotation, it is built around the mercenary enemies and has some enjoyable loot. 
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
There has been some desire for more UIVs and I thought I would try my hand at it. I'm very proud of how it turned out. Some questions I have regarding balance is that i included some c2 guns, and some guaranteed c3 drops. The c3 drops are some guns and a blueprint. The c2 guns are a 1984 and a enforcer, to provide a game play reward for mercs who follow their license laws, and the c3 guns are some jackdaws for those that don't. I chose the jackdaws and 1984s because of the difficulty to get ammo for them and the rarity of use they see. The blueprint is what I wanted to inquire about within this PR, as it is for incendiary shotgun shells. As it is the only one that will appear in a given shift (more or less), I thought it would be a fun and powerful reward, as well as make searching techfabs more of a necessity for NFSD.
The story of the UIV is that a mercenary ship overloaded while in FTL, landing in the middle of the frontier. I designed it based around "meta" designs, namely that of emp shielding a primary control area. Some other flavor touches are a printer with three piles of ash (representing burned licensees), and docking ports around the central vault to give the appearance of a ship that can detach from a heavier hull section. 
## How to test
Load into the game and use addgamerule BluespaceMercenaryFTLCrash

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
![MERCUIV FULLBRIGHT](https://github.com/user-attachments/assets/9807207c-68c9-418d-9a75-dc475bc90863)
![MERCUIV LIGHTS](https://github.com/user-attachments/assets/d9553e6c-f36a-4de9-8b50-58a07b07201c)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X ] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
:cl:
- add: Added a new Mercenary UIV.

dvir001 commented on Resources/Prototypes/_NF/Events/nf_bluespace_grids_events.yml:
```diff
@@ -299,3 +299,41 @@
+          inherent: true
+        - type: IFF
+          color: "#E10F9B"
+        - type: Shuttle
```
Shuttle?

assignedgod replied:
Ah, I didn't see that, I'll get on fixing it

dvir001 commented on Resources/Maps/_NF/Bluespace/mercuiv.yml:
```diff
@@ -0,0 +1,8243 @@
+    - type: Transform
+      pos: 1.5,3.5
+      parent: 1
+- proto: CrateEngineeringFoamGrenade
```
Fuck no

assignedgod replied:
Will remove, I added it initially because of its use in expeds/other uivs for blocking turret fire.

dvir001 commented on Resources/Maps/_NF/Bluespace/mercuiv.yml:
```diff
@@ -0,0 +1,8243 @@
+    - type: Transform
+      pos: 0.5,0.5
+      parent: 1
+- proto: NFBlueprintBoxShotgunIncendiary
```
Is this a loot pinata?

assignedgod replied:
This I added because I thought the uiv would need something unique to make it appealing, similar to how other UIVs have unique loot you can more often than not also get from them (Eg. Blood red hardsuits, cult cat, rgb staff), is there anything you suggest?

assignedgod replied:
Ah, would it be better if instead of using the blueprint, put a spawner for random incendiary ammo, making it still desirable and unique, but limited at the same time?

dvir001 reviewed (changes requested):
Avoid the loot pinata, use random spawners. this look and feels like "This is what I want to loot" then a real random event.

dvir001 commented:
All And all use random spawners or containers that has random rewards when possible, you can also add new rewards spawners if they are logical to the event.

The idea is just to avoid people looking at the map and going "oh cool I can get a double energy sword" vs oh cool tier 3/4 guns event.

assignedgod commented:
Ah, I see! So, random than using the traditional clutter spawner for mercs, I shall make some hand made ones that provide the actual things a merc would want, with rare chances at the ones they REALLY want? Easily doable, I'll get on it!

assignedgod commented:
Alrighty, I added custom loot tables and updated the UIV accordingly. I wanted to know your thoughts on the 'merc uiv rare guns' item I used to give syndicate t5 a smaller spawn chance in rare gun spawns. I really liked the idea of 'oh, cool tier 3/4 guns event' as a way to look at this, I also added some t4 melee weapon drop points in the dungeon for oni characters to get some use out of 

> All And all use random spawners or containers that has random rewards when possible, you can also add new rewards spawners if they are logical to the event.
> 
> The idea is just to avoid people looking at the map and going "oh cool I can get a double energy sword" vs "oh cool tier 3/4 guns event".
> 
> As we want the events to not end of as before when the centcom one for example was the "centcomm all channels key" event.

dvir001 commented:
You should probably use the new spawners like the wreck / vgroid ones that have more support to use existing lists so you dont need to add new lists that are a copy paste

assignedgod commented:
I'm a little confused, you said I could make new ones? I added the lists specifically to use merc items and give specific things, if you check the code I added comments explaining my choices on some of them. I specifically wanted to have less syndicate loot in the drops, and to focus a lot on t3/t4 guns. I wanted it to feel different than the regular VGroid/Wreck loot tables as the they are very generalized and I wanted this to feel different than a vgroid. 

> You should probably use the new spawners like the wreck / vgroid ones that have more support to use existing lists so you dont need to add new lists that are a copy paste

dvir001 commented:
> I'm a little confused, you said I could make new ones? I added the lists specifically to use merc items and give specific things, if you check the code I added comments explaining my choices on some of them. I specifically wanted to have less syndicate loot in the drops, and to focus a lot on t3/t4 guns. I wanted it to feel different than the regular VGroid/Wreck loot tables as the they are very generalized and I wanted this to feel different than a vgroid.
> 
> > You should probably use the new spawners like the wreck / vgroid ones that have more support to use existing lists so you dont need to add new lists that are a copy paste

I mean like the code on vgroids spawners new loot tables, you are using the old way of loot tables.

Jakumba commented:
Really neat shape and concept! 

Feedback wise, I think the inner 'vault' could use some rejigging. I'd probably make the only way in via the top access  by the bridge. At present I think it's too easy to make a beeline for the southern access, loot the armoury for some nice kit and then blaze into the vault with your new toys. 
Where as going north gets you basically nothing other than the ammo fab and a harder fight. Might be worth moving the vault doors onto the side of the box as well, so there's more space players have to traverse/fight through.

Finally, I haven't pulled it to walk through yet, but it's probably worth setting access perms on the armoury and vault exterior & interior doors to something players won't have access to, so they need to breach or hack the doors. This puts it in line with the syndie cruiser that needs an external breach point.

Over all though, this is super cool and I'm excited to see where it evolves.

Tych0theSynth commented:
While they do look cool and make for a far more interesting shape, I am concerned about the prevalence of diagonal corridors here.

Diagonal walls are notoriously scrunkly when it comes to a whole host of things - namely atmos and physics interactions, as well as making piping look horrible.

I'm concerned that these could be an annoying nightmare to fight in, especially against mercs. I'm curious, have you spawned this and cleared it yourself?

I'm also curious as to the particular choice of shape? Why this? What are the mercs doing there and why does it look like that instead of something more classically ship-like? (This particular bit isn't criticism, I'm just interested in your design philosophy here).

Jakumba commented:
> While they do look cool and make for a far more interesting shape, I am concerned about the prevalence of diagonal corridors here.
> 
> Diagonal walls are notoriously scrunkly when it comes to a whole host of things - namely atmos and physics interactions, as well as making piping look horrible.
> 
> I'm concerned that these could be an annoying nightmare to fight in, especially against mercs. I'm curious, have you spawned this and cleared it yourself?
> 
> I'm also curious as to the particular choice of shape? Why this? What are the mercs doing there and why does it look like that instead of something more classically ship-like? (This particular bit isn't criticism, I'm just interested in your design philosophy here).

at the risk of speeding for her, I got the vibe it was meant to be a really big Gourd :p

assignedgod commented:
> > I'm a little confused, you said I could make new ones? I added the lists specifically to use merc items and give specific things, if you check the code I added comments explaining my choices on some of them. I specifically wanted to have less syndicate loot in the drops, and to focus a lot on t3/t4 guns. I wanted it to feel different than the regular VGroid/Wreck loot tables as the they are very generalized and I wanted this to feel different than a vgroid.
> > > You should probably use the new spawners like the wreck / vgroid ones that have more support to use existing lists so you dont need to add new lists that are a copy paste
> 
> I mean like the code on vgroids spawners new loot tables, you are using the old way of loot tables.

Ah! I see. I will get on fixing that ASAP

assignedgod commented:
> Really neat shape and concept!
> 
> Feedback wise, I think the inner 'vault' could use some rejigging. I'd probably make the only way in via the top access by the bridge. At present I think it's too easy to make a beeline for the southern access, loot the armoury for some nice kit and then blaze into the vault with your new toys. Where as going north gets you basically nothing other than the ammo fab and a harder fight. Might be worth moving the vault doors onto the side of the box as well, so there's more space players have to traverse/fight through.
> 
> Finally, I haven't pulled it to walk through yet, but it's probably worth setting access perms on the armoury and vault exterior & interior doors to something players won't have access to, so they need to breach or hack the doors. This puts it in line with the syndie cruiser that needs an external breach point.
> 
> Over all though, this is super cool and I'm excited to see where it evolves.

Yeah, I definitely want to tweak the layout a bit, I changed some of the boss and loot drops around and need to update the screenshots, first I just want to get the loot spawners all finished so I don't have to keep stitching images together. I'll definitely be taking that idea, though, as it'll let me sequester away even more of the ship's primary functions into little side compartments, and would make sense with how the ship is meant to be

assignedgod commented:
And lastly the diagonal walls were VERY rough when I tested it, I was just thinking about how I wanted to fix it. I think what I might do is add short hallways to small transition rooms on the corners and keep the general footprint with the design by using more external walls

assignedgod commented:
> > I'm a little confused, you said I could make new ones? I added the lists specifically to use merc items and give specific things, if you check the code I added comments explaining my choices on some of them. I specifically wanted to have less syndicate loot in the drops, and to focus a lot on t3/t4 guns. I wanted it to feel different than the regular VGroid/Wreck loot tables as the they are very generalized and I wanted this to feel different than a vgroid.
> > > You should probably use the new spawners like the wreck / vgroid ones that have more support to use existing lists so you dont need to add new lists that are a copy paste
> 
> I mean like the code on vgroids spawners new loot tables, you are using the old way of loot tables.

I'm just curious because I'm having a lot of trouble with and want to get clarification before I redo the tables, whats the benefit of the new way the tables are done vs the old ways the tables are done? The main reason I used the old way is because the syndicate UIV used the old method of doing tables, and it was easy to wrap my head around as I'm pretty new at this. Is it just that the new way will be more maintained in the future? I just wanted to check before I redid the tables for what will essentially be (if I am reading the code correctly) the same/a similar result.

assignedgod commented:
I have yet to figure out how the scrap/vgroid tables work, I've mostly just been trying to figure that out. I'm taking a little break from this, so I'm uploading the grid as is in case anyone wants to test it themselves.

assignedgod commented:
Alright, everything Seems to be working, and I got the new loot tables to work. I added a folder to hold UIV spawners, and got the spawners working. Let me know if it needs any tweaks or balance changes, seemed alright when I ran through it but I may have missed something

dvir001 commented:
To avoid having to micromanage the same spawners 2 times @ErhardSteinhauer please convert when you can the expo spawners to new format so we can combine this to not upkeep 2 of the same pools

ErhardSteinhauer commented:
> To avoid having to micromanage the same spawners 2 times @ErhardSteinhauer please convert when you can the expo spawners to new format so we can combine this to not upkeep 2 of the same pools

Yeah, gonna be a while before that happens though: there are a lot of spawners there

assignedgod commented:
So, are the reward tables good? The UIV is functional, am I just waiting on the loot tables update for this to be reviewed or is there more work I need to do for this?

assignedgod commented:
![Mercuiv lit up](https://github.com/user-attachments/assets/48af0820-40a9-4215-8bd0-6ba2e167a905)
![mercuiv fullbright(2)](https://github.com/user-attachments/assets/4c485d03-5087-4de6-8903-b58255585677)
![image](https://github.com/user-attachments/assets/78542422-9452-4f06-a21a-992118ff2b33)

Some changes to the structure to make it look better on the mass scanner

assignedgod commented:
Oh shit whoops, didn't know it would do that
